### PR TITLE
platform: update xtensa headers for BYT/CHT and remove GCC W/A for APL

### PR DESCRIPTION
**Pull request by tlauda**
Updates xtensa headers for BYT/CHT platforms and removes previously added workaround for APL GCC compilation.

Requires https://github.com/thesofproject/xtensa-overlay/pull/3.
Requires https://github.com/thesofproject/xtensa-overlay/pull/4.

### CODE REVIEW

**Comment by jajanusz** [comment]
@tlauda is it for #1401 ?

**Comment by tlauda** [comment]
> @tlauda is it for #1401 ?

@jajanusz It depends, how to look at it, but basically thesofproject/xtensa-overlay#3 is for #1401. This one just requires the new overlay to fix the original issue.

**Comment by jajanusz** [comment]
I guess we need @xiulipan to add https://github.com/thesofproject/xtensa-overlay/pull/3 to CI then merge it.

**Comment by xiulipan** [comment]
@jajanusz 
I will try to rebuild the cross compiler after the fix is merge

**Comment by xiulipan** [comment]
@tlauda @jajanusz 
I have update the Docker image for CI, please rebase and update the PR also.

**Comment by tlauda** [comment]
SOFCI TEST

**Comment by tlauda** [comment]
@lgirdwood It looks like byt and cht have some problems with `xchal_cp1_store` and `xchal_cp1_load`. Are we sure that xtensa headers in platform are correct?

**Comment by tlauda** [comment]
@jajanusz @lgirdwood Needed to also update byt/cht headers and overlay. @xiulipan You need to use xtensa-byt-overlay from https://github.com/thesofproject/xtensa-overlay/pull/4.

**Comment by xiulipan** [comment]
@tlauda 
Will try to make a update for docker image ASAP.

**Comment by xiulipan** [comment]
@tlauda
I have updated the docker, hope the build will pass now.

**Comment by xiulipan** [comment]
SOFCI TEST

**Comment by xiulipan** [comment]
SOFCI TEST

**Comment by tlauda** [comment]
@lgirdwood @xiulipan It looks like there is still a problem with coprocessor store on BYT, even with the correct definitions. Maybe I'm missing something. I will look into it later.

File: rimage/file_simple.c [hunk 1]
```diff
@@ -13,10 +13,10 @@
 
 #define BYT_IRAM_BASE		0xff2c0000
 #define BYT_IRAM_HOST_OFFSET	0x0C0000
-#define BYT_IRAM_SIZE		(80 * 1024)
+#define BYT_IRAM_SIZE		(8 * 32 * 1024)
```
Comment (lgirdwood) [inline rimage/file_simple.c]: Probably worth double checking this, it maybe that the address space is 256kB but only 80kB is populated ? The docs I have all showed the smaller size.

Comment (plbossart) [reply rimage/file_simple.c]: yes, there's only 80 KB, why would you want to change this setting for an unrelated GCC compilation issue on a different platform anyways?

Comment (plbossart) [reply rimage/file_simple.c]: I just double-checked the HAS, it's 80 KB IRAM, 160 KB Data RAM, 48 KB I$, 96 KB D$

No idea how the 256 KB reference came from.

**Comment by tlauda** [comment]
@lgirdwood For some reason coprocessor store/load still doesn't work for non-cAVS platforms. Since we can't use them with GCC anyway, I've decided to leave them only for XCC compiled versions.

@mengdonglin Please run some basic tests on BYT with this set of patches. I don't expect any regression, but better be sure.

**Pull request merged**